erts/emulator/beam/erl_init.c

```diff
@@ -793,6 +941,9 @@ early_init(int *argc, char **argv) /*
     char envbuf[21]; /* enough for any 64-bit integer */
     size_t envbufsz;
     int skip_read_topology = 0;
+#if defined(__linux__) && defined(__x86_64__) && defined(__LP64__)
```

Comment:
Sure, I can move this into `configure`.  This is limited to 64-bit x86 since 32-bit x86 has a 4MiB alternate page size, inherited from the i860, which you can read about [here](http://www.rcollins.org/articles/4mpages/4MOverview.html).

Outside of x86, the sizes of the alternate page sizes varies a lot.  64-bit ARM on Linux has 4KiB 2MiB and 1GiB pages like the x86.  32-bit ARM has, IIRC, 1MiB pages.  64-bit ARM with a 64KiB page has 512MiB pages.  PowerPC has 64KiB, 16MiB, pages, at least.  It's a zoo.